### PR TITLE
General documentation updates

### PR DESCRIPTION
**Pull request by IgorSusmelj**
### Changes

- Update some docstrings (tried to make them more descriptive)
- Update documentation (add more links and add missing pieces)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,7 +64,7 @@ Lightly requires **Python 3.6+**. We recommend installing Lightly in a **Linux**
 
 - hydra-core>=1.0.0
 - numpy>=1.18.1
-- pytorch_lightning>=0.10.0   
+- pytorch_lightning>=1.0.4
```
Comment (philippmwirth) [inline README.md:67]: There are now more dependencies, checkout out the base.txt file:

```
hydra-core>=1.0.0
numpy>=1.18.1
pytorch_lightning>=1.0.4
requests>=2.23.0
torchvision
tqdm>=4.44
certifi >= 14.05.14
six >= 1.10
python_dateutil >= 2.5.3
setuptools >= 21.0.0
urllib3 >= 1.15.1
```

Comment (philippmwirth) [reply README.md:67]: Obviously you don't have to list all of them. E.g. urllib3 is also a dependency of requests.

File: docs/README.md [hunk 2]
```diff
@@ -13,6 +13,11 @@ pip install sphinx_rtd_theme
 make html
 ```
 
+Shortcut to build the docs (with env variables for active-learning tutorial) use:
+```
+LIGHTLY_SERVER_LOCATION='https://api.lightly.ai' TOKEN='YOUR_TOKEN' AL_TUTORIAL_DATASET_ID='YOUR_DATASET_ID' make html && python -m http.server 1234 -d build/html
```
Comment (japrescott) [inline docs/README.md:18]: do you always want to spcify the LIGHLLY_SERVER_LOCATION?

File: docs/source/docker/getting_started/first_steps.rst [hunk 3]
```diff
@@ -124,6 +124,14 @@ The command above does the following:
   will be 30% of the initial dataset size. You can also specify the exact 
   number of remaining images by setting **n_samples** to an integer value.
 
+You can also use the stopping condition **min_distance**. This allows you specify
```
Comment (philippmwirth) [inline docs/source/docker/getting_started/first_steps.rst:127]: There's a typo here: Final -> final.

Also, I think the wording is a little rough. What about

"This allows you to specify the minimum allowed distance between two image embeddings in the output dataset. After normalizing the input embeddings to unit length, this value should be between 0 and 2."

File: docs/source/docker/getting_started/first_steps.rst [hunk 4]
```diff
@@ -124,6 +124,14 @@ The command above does the following:
...
+a minimum distance of the normalized embeddings you want to keep in the Final
+dataset. This is often a more convenient method when working with different data
+sources and trying to combine them in a balanced way.
+
+- **stopping_condition.min_distance=0.2** would remove all samples which are
```
Comment (philippmwirth) [inline docs/source/docker/getting_started/first_steps.rst:132]: ...closer to each other than

File: docs/source/docker/overview.rst [hunk 5]
```diff
@@ -12,8 +12,16 @@ and an easy way to work with lightly. But there is more!
 With the introduction of our on-premise solution, you can process larger datasets completely on your end without data leaving your infrastructure.
 We worked hard to make this happen and are very proud to present you with the following specs:
 
+* **NEW** Lightly Docker has built-in pretagging models (see :ref:`ref-docker-pretagging` )
+
+  * Use this feature to pre-label your dataset or to only select images with a certain object
+
+  * Support object categories are: bicycle, bus, car, motorcycle, person, train, truck
```
Comment (philippmwirth) [inline docs/source/docker/overview.rst:19]: Support**ed**

File: docs/source/docker/overview.rst [hunk 6]
```diff
@@ -12,8 +12,16 @@ and an easy way to work with lightly. But there is more!
 With the introduction of our on-premise solution, you can process larger datasets completely on your end without data leaving your infrastructure.
 We worked hard to make this happen and are very proud to present you with the following specs:
 
+* **NEW** Lightly Docker has built-in pretagging models (see :ref:`ref-docker-pretagging` )
+
+  * Use this feature to pre-label your dataset or to only select images with a certain object
```
Comment (philippmwirth) [inline docs/source/docker/overview.rst:17]: images which contain certain objects

File: docs/README.md (hunk 2 above)
Comment (IgorSusmelj) [reply docs/README.md:18]: This is what I use to run it. You can also export the variable. I prefer setting it every time so I don't accidentally have all my lightly stuff connecting to the dev API whereas I want to run on the prod one :)

File: docs/source/getting_started/active_learning.rst [hunk 7]
```diff
@@ -1,22 +1,26 @@
 .. _lightly-active-learning:
 
-Active Learning
+Active-Learning
```
Comment (philippmwirth) [inline docs/source/getting_started/active_learning.rst:3]: Are you sure about the hyphen? I've never seen that anywhere before.

File: docs/source/index.rst [hunk 8]
```diff
@@ -20,6 +20,12 @@ Lightly has been built to help you understand and work with large unlabeled
 datasets. It is built on top of PyTorch and therefore fully compatible with 
 other frameworks such as Fast.ai.
 
+**NEW** Lightly has now integrated support for active-learning in combination
```
Comment (philippmwirth) [inline docs/source/index.rst:23]: Lightly now has

File: lightly/data/dataset.py [hunk 9]
```diff
@@ -93,26 +93,50 @@ class LightlyDataset:
...
         transform:
             Image transforms (as in torchvision).
         index_to_filename:
             Function which takes the dataset and index as input and returns
             the filename of the file at the index. If None, uses default.
 
     Examples:
-        >>> # load cifar10 from a local folder
+        >>> # load a dataset consisting of images from a local folder
+        >>> # mydata
```
Comment (philippmwirth) [inline lightly/data/dataset.py:120]: Maybe add a slash here? `>>> # mydata/`
If you do it, also do it with the next three examples.

File: lightly/models/simclr.py [hunk 10]
```diff
@@ -24,7 +24,11 @@ def _get_simclr_projection_head(num_ftrs: int, out_dim: int):
 
 
 class SimCLR(nn.Module):
-    """Implementation of the SimCLR architecture.
+    """Implementation of the SimCLR[0] architecture
+
+    Recommended loss: :py:class:`lightly.loss.ntx_ent_loss.NTXentLoss`
```
Comment (philippmwirth) [inline lightly/models/simclr.py:29]: Nice, good idea :)

**Review by philippmwirth — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (IgorSusmelj) [reply README.md:67]: I'm not sure I would include them. I would rather wish to check whether we really need all of them. 
I raised the question already once. I don't see why a lightly user needs setuptools^^
Let's also fix this in this sprint. I'll create a follow-up issue.

Comment (IgorSusmelj) [reply README.md:67]: Follow up issue is here: https://github.com/lightly-ai/lightly/issues/219

File: docs/source/docker/getting_started/first_steps.rst (hunk 3 above)
Comment (IgorSusmelj) [reply docs/source/docker/getting_started/first_steps.rst:127]: I like

File: docs/source/getting_started/active_learning.rst (hunk 7 above)
Comment (IgorSusmelj) [reply docs/source/getting_started/active_learning.rst:3]: Hmm, I thought it was with hyphen :S
Now that I checked everywhere, you're right. Will undo

File: docs/source/docker/overview.rst [hunk 11]
```diff
@@ -14,14 +14,14 @@ We worked hard to make this happen and are very proud to present you with the fo
...
+  * Use this feature to pre-label your dataset or to only select images which contain certain objects
 
-  * Support object categories are: bicycle, bus, car, motorcycle, person, train, truck
+  * Supported object categories are: bicycle, bus, car, motorcycle, person, train, truck
 
 * Sample more than 1 Million samples within a few hours!
 
 * Runs directly with videos without prior extraction of the frames!
-
+ƒ
```
Comment (philippmwirth) [inline docs/source/docker/overview.rst:24]: typo?

**Review by philippmwirth — APPROVED** [review]
